Data/Yaml/Pretty.hs

```diff
@@ -0,0 +1,38 @@
+-- | Prettier YAML encoding.
+module Data.Yaml.Pretty
+    ( encodePretty
+    , Config(..)
```

Comment:
In order to avoid API breakage, I'd prefer:

* Don't expose the constructor
* Expose `setConfCompare` as a new setter function

This is the approach we use in warp and others, and is spelled out in more detail at https://github.com/commercialhaskell/haskelldocumentation/blob/master/content/designing-apis-for-extensibility.md